### ISSUE TITLE
Overfitting on "iteration" parameters?

### ISSUE THREAD

**Issue by firrice** [MalongTech/research-ms-loss#36]
Hi, the unify framework for all knids of paired-loss proposed in the paper is great, while i found that it appeared that the best "test recall" has been actually decided by val_dataset, which refenced to the raw code below:
![image](https://github.com/msight-tech/research-ms-loss/assets/26590498/8fff44b0-8e1d-47f6-ada9-12de5a495581)
According to the fig above, "val datatset" actually also plays a role of "test dataset", which means "test dataset" is visible during training. 
 So does it seems like choosing a "best train iteration" parameter, which is a risk of overfitting on training hyperparameters? 
(I have found similar operation in several other papers, and i knew there was a lack of test dataset building the dataset, such as the general protocal "construct query+gallery based on the raw val+test split in DeepFashion")
